eth2near/eth2near-block-relay-rs/src/contract_type.rs

```diff
@@ -0,0 +1,59 @@
+use std::error::Error;
+use std::fmt;
+use std::fmt::{Display, Formatter};
+use std::str::FromStr;
+use serde::Deserialize;
+
+#[derive(Debug, Clone, Deserialize)]
+pub enum ContractType {
+    Near,
+    Dao,
+    File,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct IncorrectContractType;
+
+impl Display for IncorrectContractType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "Unknown contract type. Possible contract types: 'Near', 'Dao', 'File'"
+        )
+    }
+}
+
+impl Error for IncorrectContractType {}
+
+impl Display for ContractType {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            ContractType::Near => write!(f, "Near"),
+            ContractType::Dao => write!(f, "Dao"),
+            ContractType::File => write!(f, "File"),
+        }
+    }
+}
+
+impl ContractType {
+    pub fn as_str(&self) -> &str {
+        match self {
+            ContractType::Near => "Near",
+            ContractType::Dao => "Dao",
+            ContractType::File => "File",
+        }
+    }
+}
+
+impl FromStr for ContractType {
+    type Err = IncorrectContractType;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "Near" => Ok(ContractType::Near),
+            "Dao" => Ok(ContractType::Dao),
+            "File" => Ok(ContractType::File),
+            _ => Err(IncorrectContractType),
+        }
```

Comment:
It's better to parse the lowercase version, which allows more invariants.

```suggestion
        match s.to_lowercase().as_str() {
            "near" => Ok(ContractType::Near),
            "dao" => Ok(ContractType::Dao),
            "file" => Ok(ContractType::File),
            _ => Err(IncorrectContractType),
        }
```

Reply:
fixed 
https://github.com/aurora-is-near/rainbow-bridge/pull/825/commits/17bcda0b6dc5e0fd8048a69831817d232213b0ad